Jint/Native/Array/ArrayInstance.cs

```diff
@@ -358,9 +372,13 @@ public sealed override List<JsValue> GetOwnPropertyKeys(Types types = Types.None
                 var length = System.Math.Min(temp.Length, GetLength());
                 for (var i = 0; i < length; i++)
                 {
-                    var descriptor = temp[i];
-                    if (descriptor != null)
+                    var value = temp[i];
+                    if (value != null)
                     {
+                        if (value is not PropertyDescriptor descriptor)
```

Comment:
Would it be better to do `if (value is JsValue jsValue)`, at least you can then resue `jsValue` directly so this statement's result is 'reused'. Here you only take advantage of the variable declaration for `descriptor`.

Reply:
We need to return `PropertyDescriptor` instances so my reasoning here was to materialize it if it wasn't present already, then it will be used in subsequent calls. This is not a common thing to be called, mostly JSON serialization and `Object.getOwn*` functionality I think.